caterva2/tests/test_api.py

```diff
@@ -76,17 +76,32 @@ def test_lazyexpr(services, sub_urlbase, sub_jwt_cookie):
                            auth_cookie=sub_jwt_cookie)
     assert lxpath == f'@scratch/{lxname}.b2nd'
 
+    expression2 = f'{opnm} * 2'
+    operands2 = {opnm: lxpath}
+    lxname = 'expr_from_expr'
+    lxpath2 = cat2.lazyexpr(lxname, expression2, operands2, sub_urlbase, auth_cookie=sub_jwt_cookie)
+    assert lxpath2 == f'@scratch/{lxname}.b2nd'
+
     # Check result metadata.
     lxinfo = cat2.get_info(lxpath, sub_urlbase, auth_cookie=sub_jwt_cookie)
-    assert lxinfo['shape'] == opinfo['shape']
-    assert lxinfo['dtype'] == opinfo['dtype']
+    lxinfo2 = cat2.get_info(lxpath2, sub_urlbase, auth_cookie=sub_jwt_cookie)
+    assert lxinfo['shape'] == opinfo['shape'] == lxinfo2['shape']
+    assert lxinfo['dtype'] == opinfo['dtype'] == lxinfo2['dtype']
     assert lxinfo['expression'] == f'({expression})'.replace(opnm, 'o0')
+    assert lxinfo2['expression'] == '((o0 + 1) * 2)'
     assert lxinfo['operands'] == dict(o0=operands[opnm])
+    assert lxinfo2['operands'] == dict(o0=operands[opnm])
+
+    # Get one chunk
+    _ = cat2.get_chunk(lxpath2, 0, sub_urlbase, auth_cookie=sub_jwt_cookie)
+    _ = cat2.get_chunk(lxpath, 0, sub_urlbase, auth_cookie=sub_jwt_cookie)
 
     # Check result data.
     a = cat2.fetch(oppt, sub_urlbase, auth_cookie=sub_jwt_cookie)
-    b = cat2.fetch(lxpath, sub_urlbase, auth_cookie=sub_jwt_cookie)
-    np.testing.assert_array_equal(a[:], b[:])
+    b = cat2.fetch(lxpath2, sub_urlbase, auth_cookie=sub_jwt_cookie)
+    c = cat2.fetch(lxpath, sub_urlbase, auth_cookie=sub_jwt_cookie)
+    np.testing.assert_array_equal((a[:] + 1) * 2, b[:])
```

Comment:
I'd prefer to invert the roles of `b` and `c` here. It seems to me that it follows the code flow above more closely.  Other than that, LGTM.

Reply:
Also, it would be nice to keep the original test and add another one.  In case something breaks, it typically helps to know whether all the tests fail, or only the more complex ones.

Reply:
And I'd add a new test where two equal expressions are evaluated separately, and then check that the retrieved chunks are equal.  That would be an easy way to test `get_chunk`.

Reply:
Fixed in 95a976c3325d111544d132804d0d3ce8b0eddca3.